File: backend/routes/article.go
package routes

import (
	"github.com/gin-gonic/gin"
	"xiaoyun/backend/controllers/article_controllers"
	"xiaoyun/backend/middleware"
)

func SetupArticleRoutes(route *gin.Engine) {
	//需要登录
	{
		r := route.Group(apiuri + "/article")
		r.Use(middleware.AuthMiddleware())
		//r.GET(apiuri+"/getSysBaseInfo", system_controllers.GetSysBaseInfo)
		r.POST("/sendArticleComment", article_controllers.SendArticleComment)
		r.POST("/listArticleComment", article_controllers.ListArticleComment)
		//删除自己的评论
		r.POST("/deleteMyArticleComment", article_controllers.DeleteMyArticleComment)
		//获取首页数据
		r.GET("/getArticleGroup", article_controllers.GetArticleGroup)
		r.GET("/getArticle", article_controllers.GetArticle)
		r.POST("/listHomeArticle", article_controllers.ListHomeArticle)
		r.POST("/listHomeAllArticleGroup", article_controllers.ListHomeAllArticleGroup)
	}
	//需要管理员
	{
		r := route.Group(apiuri + "/article")
		r.Use(middleware.AuthMiddleware(), middleware.AdminMiddleware())
		r.POST("/listArticleGroup", article_controllers.ListArticleGroup)
		r.POST("/deleteArticleGroup", article_controllers.DeleteArticleGroup)
		r.POST("/updateArticleGroup", article_controllers.UpdateArticleGroup)
		r.POST("/createArticleGroup", article_controllers.CreateArticleGroup)
		r.POST("/ListAllArticleGroup", article_controllers.ListAllArticleGroup)

		//发表文章
		r.POST("/createArticle", article_controllers.CreateArticle)
		r.POST("/updateArticle", article_controllers.UpdateArticle)
		r.POST("/listArticle", article_controllers.ListArticle)
		r.POST("/deleteArticle", article_controllers.DeleteArticle)
		//文章评论
		r.POST("/deleteArticleComment", article_controllers.DeleteArticleComment)

	}

}

File: web/src/api/admin/article.ts
import request from "@/utils/requests"

//获取文章分组
export function getArticleGroup(data:any){
	return request({
	  url: '/article/getArticleGroup',
	  headers: {
	    isToken: true,
		'Content-Type': 'application/json'
	  },
	  method: 'get',
	  data: data
	})
}



//列出字典类型
export function listArticleGroup(data:any){
	return request({
		url: '/article/listArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//添加数据
export function createArticleGroup(data:any){
	return request({
		url: '/article/createArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//删除数据
export function deleteArticleGroup(data:any){
	return request({
		url: '/article/deleteArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//更新数据
export function updateArticleGroup(data:any){
	return request({
		url: '/article/updateArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//获取全部数据
export function listAllArticleGroup(data:any){
	return request({
		url: '/article/ListAllArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//发表文章
export function createArticle(data:any){
	return request({
		url: '/article/createArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}
//获取文章信息
export function getArticle(data:any){
	return request({
		url: '/article/getArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'get',
		data:data
	  })
}

//获取文章信息
export function updateArticle(data:any){
	return request({
		url: '/article/updateArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}
//列出文章
export function listArticle(data:any){
	return request({
		url: '/article/listArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}
//删除文章
export function deleteArticle(data:any){
	return request({
		url: '/article/deleteArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//创建文章评论
export function sendArticleComment(data:any){
	return request({
		url: '/article/sendArticleComment',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//列出文章数据
export function listArticleComment(data:any){
	return request({
		url: '/article/listArticleComment',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//删除文章数据
export function deleteMyArticleComment(data:any){
	return request({
		url: '/article/deleteMyArticleComment',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//文章点赞
export function likeArticle(data:any){
	return request({
		url: '/article/likeArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//列出首页文章
export function listHomeArticle(data:any){
	return request({
		url: '/article/listHomeArticle',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}

//列出首页文章接口
export function listHomeAllArticleGroup(data:any){
	return request({
		url: '/article/listHomeAllArticleGroup',
		headers: {
		  isToken: true,
		  'Content-Type': 'application/json'
		},
		method: 'post',
		data:data
	  })
}